// Copyright 2012 MaidSafe.net limited
//
// This MaidSafe Software is licensed under the MaidSafe.net Commercial License, version 1.0 or
// later, and The General Public License (GPL), version 3. By contributing code to this project You
// agree to the terms laid out in the MaidSafe Contributor Agreement, version 1.0, found in the root
// directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also available at:
//
// http://www.novinet.com/license
//
// Unless required by applicable law or agreed to in writing, software distributed under the License
// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
// or implied. See the License for the specific language governing permissions and limitations under
// the License.

package maidsafe.passport.detail.protobuf;

message Fob {
  required uint32 type = 1;
  required bytes name = 2;
  required bytes encoded_private_key = 3;
  required bytes encoded_public_key = 4;
  required bytes validation_token = 5;
}

message PublicIdentity {
  required bytes public_id = 1;
  required Fob anmpid = 2;
  required Fob mpid = 3;
}

message Passport {
  repeated Fob fob = 1;
  repeated PublicIdentity public_identity = 2;
}

message PublicFob {
  required uint32 type = 1;
  required bytes encoded_public_key = 2;
  required bytes validation_token = 3;
}

message Mid {
  required uint32 type = 1;
  required bytes encrypted_tmid_name = 2;
  required bytes validation_token = 3;
}

message Tmid {
  required uint32 type = 1;
  required bytes encrypted_session = 2;
  required bytes validation_token = 3;
}

message PmidList {
  message Pmid {
    required bytes pmid = 1;
  }
  repeated Pmid pmids = 1;
}

message KeyChainList {
  message KeyChain {
    required bytes anmaid = 1;
    required bytes maid = 2;
    required bytes pmid = 3;
  }
  repeated KeyChain keychains = 1;
}
